### PR TITLE
Add stop/resume methods and checkpoint object

### LINKED ISSUE #213: Training stop/resume and checkpointing
## Feature Description
With Training API (#212 ) in place, we can add ability to stop training and save intermediate training info to resume training later.

```
// Start the training
// Training object would contain current epoch, batch, modelParameters
training = Job.train(...)
```

Suggested API:
```
// Stop training
training.stop()
```

New events in Job.train: `'stop'`

```
// User-defined serialization (serialize/unserialize/storage is up to user)
serialized_checkpoint = serialize(training)
unserialized_checkpoint = unserialize(serialized_checkpoint)

// Supplying checkpoint back to Job.train
training = Job.train(trainingPlan, {
   ...
   checkpoint: unserialized_checkpoint
})
```

Training loop should read properties of checkpoint 
and load model params, epoch, step, batchSize, etc. from it.

## What alternatives have you considered?
API was discussed in FL team.

## Additional Context
See #212

### PR DESCRIPTION
**Pull request by vvmnnnkv**
## Description
Fixes #213

## Affected Dependencies
n/a

## How has this been tested?
* added unit tests
* tested with MNIST example

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,7 +195,71 @@ const outputs = [
...
+
+```javascript
+const checkpoint = await training.stop();
+const checkpointJson = await checkpoint.toJSON();
+localStorage.setItem('checkpoint', JSON.stringify(checkpointJson));
+
+// ... checkpoint can survive page reload ...
+
+const checkpointJsonString = localStorage.getItem('checkpoint');
+const checkpointJson = JSON.parse(checkpointJsonString);
```
Comment (cereallarceny) [inline README.md:240]: You should make this step internal to the fromJSON method. No need to get the user to do this.

**Review by cereallarceny — CHANGES_REQUESTED** [review]
Literally one change

**Review by cereallarceny — APPROVED** [review]
